### DISCUSSION TITLE
A way to change default Krita AI Docker settings?

### DISCUSSION THREAD

**Discussion by GentlemanRevvnar** [Acly/krita-ai-diffusion#902] [Q&A]
Long story short i primarly use this plugin to tweak minor details of existing images.
Whenever I go into the Docker I **always** have to: 
switch to my own custom style (from the default 1.5 Cinematic Photo), 
then adjust the generation strenght from 100% to ~55/65%, 
then change Refine type to custom, 
then disable seamless, 
then change automatic context to selection bounds, 
then change batchcount to 3
...
This might not seem much, but editing 200 different images, and having to change these settings back (from default ones) between each image is extremely tedious.
I couldn't find any option to change the default state of these settings upon opening any picture in Krita, so I'd like to ask if  this is even possible to do? I should also propably mention that i'm illiterate when it comes to programming anything more than changing a few lines in existing code 💀

Tl:dr how do i change default settings of the Krita AI Docker?
![image](https://github.com/Acly/krita-ai-diffusion/assets/100436143/c370ab7e-e575-4b3a-b896-5bf44152d0da)

**Comment by minsky91** [comment]
I think yours is a reasonable question, I could also relate to it. But a feature to manage a complex set of settings could be a tricky thing to implement, I imagine. In the meantime, here's a suggestion. To me, it sounds like you basically have just a single workflow/project with a varying input component (the image). Krita, together with the plugin, keeps a lot of context in the .kra document, various settings get saved and restored with it. You could try I suppose to keep a single document for your project, flatten your image after you are done processing it (if in your workflow you create layers, that is) and just replace the background one with the next image to be processed. If you keep doing this without closing the document, your settings will be always retained this way. If you have to close the document or restart the computer, some of your last used settings will be restored from that single .kra document you maintain for your project. Or you could clone the document for every new image to be processed.

Does this make sense to you?

**Reply by GentlemanRevvnar** [reply]
Yeah, I've been thinking about this approach, but then upon loading new image, the namefile of the project doesn't change, so having to manually Save As every image, and manually changing their appropriate names is just as tedious 😭

**Comment by Acly** [comment]
I was thinking along the same lines. Special UI seems a bit overkill. I'm making it so new document is created with the most recently used settings (for style, inpaint params, batch... not for things like prompt or strength). That should work out most of the time?

**Reply by minsky91** [reply]
Yeah, I've seen how such a feature is implemented in SwarmUI and it's terrible, with a huge dialogue box and myriads of checkboxes for the user to tick when saving to a 'style' preset - I'd rather not! Retaining of the last used settings when a new document is created would be a great time-saver for many Krita AI workflows, I personally would really appreciate it. (Most crucially, retaining the last used style and batch number.) Perhaps, with an accompanying setting to switch this feature on or off when necessary, and also making Restore Defaults more nuanced (which a choice of particular set of defaults to restore, that is). That should not take too much coding, I suppose, and won't overwhelm the UI.  

And while we are on this subject, could you please also add a hotkey/shortcut for switching the style and - especially - the checkpoint? This is the single most frequent operation I use in my work with the plugin, and I imagine many others do too. Changing of the sampler is also a pretty frequent one. All require clicking on the gear icon and then on the corresponding dropdown box, and that's a lot of mouse travel.
